src/composables/draw/drawn-features.composable.ts

```diff
@@ -1,30 +1,135 @@
+import { t } from 'i18next'
 import useMap from '../map/map.composable'
 import { Collection, Feature } from 'ol'
-import { Geometry } from 'ol/geom'
-import VectorLayer from 'ol/layer/Vector'
-import VectorSource from 'ol/source/Vector'
-
-const DEFAULT_DRAW_ZINDEX = 1000
+import { Circle, Geometry, LineString } from 'ol/geom'
+import Polygon, { fromCircle } from 'ol/geom/Polygon'
+import { storeToRefs } from 'pinia'
+import { useDrawStore } from '@/stores/draw.store'
+import { createStyleFunction } from './draw-utils'
+import { useAppStore } from '@/stores/app.store'
+import { screenSizeIsAtLeast } from '@/services/common/device.utils'
 
 export default function useDrawnFeatures() {
   const map = useMap().getOlMap()
+  const { drawStateActive, drawnFeatures } = storeToRefs(useDrawStore())
+  const { addDrawnFeature } = useDrawStore()
+  const { feedbackOpen, feedbackanfOpen, feedbackageOpen, feedbackcruesOpen } =
+    storeToRefs(useAppStore())
+  const { toggleMyMapsOpen } = useAppStore()
 
-  const features = new Collection<Feature<Geometry>>()
-  const drawLayer = new VectorLayer({
-    source: new VectorSource({
-      features,
-    }),
-    zIndex: DEFAULT_DRAW_ZINDEX,
-    // altitudeMode: 'clampToGround', //used in v3, but causes type error
-  })
-  //was done in MymapsController in v3
-  map.addLayer(drawLayer)
+  const features = drawnFeatures.value as Collection<Feature<Geometry>>
 
   function addFeature(feature: Feature<Geometry>) {
-    //todo: handle mutuability, save to mymaps, encode in URL
-    features.push(feature)
+    const name = getName(feature)
+    const nbFeatures = features.getLength()
+    feature.set('name', name + ' ' + (nbFeatures + 1))
+    feature.set('description', '')
+    feature.set('__editable__', true)
+    feature.set('color', '#ed1c24')
+    feature.set('opacity', 0.2)
+    feature.set('stroke', 1.25)
+    feature.set('size', 10)
+    feature.set('angle', 0)
+    feature.set('linestyle', 'plain')
+    feature.set('shape', 'circle')
+    feature.set('isLabel', drawStateActive.value === 'drawLabel')
+    feature.setStyle(createStyleFunction(map))
+    feature.set('display_order', nbFeatures)
```

Comment:
TODO: put all this definition in a wrapper dedicated to v4 containing these info and the ol feature geom. To be discussed together.